src/java.base/share/classes/java/net/DatagramPacket.java

```diff
@@ -409,10 +412,8 @@ public synchronized void setData(byte[] buf) {
      * @since   1.1
      */
     public synchronized void setLength(int length) {
-        if ((length + offset) > buf.length || length < 0 ||
-            (length + offset) < 0) {
-            throw new IllegalArgumentException("illegal length");
-        }
+        Preconditions.checkFromIndexSize(offset, length, buf.length,
+                Preconditions.outOfBoundsExceptionFormatter(IllegalArgumentException::new));
```

Comment:
Thanks for that. Maybe the bound checks in

```
     public synchronized void setData(byte[] buf, int offset, int length) {
```

could be replaced in a similar way.

Reply:
Hello Daniel,

I've now updated this method too to use `Preconditions`. Additionally, a `NullPointerException` check which was implicit in this method (and noted by a code comment) has now been made an explicit check (using `Objects.requireNonNull`) in this updated version. I hope that's OK.

The existing `Setters.java` test case covers this method as well and I've run it after these changes to make sure it continues to pass.